### PR TITLE
Cancel duplicate workflow runs

### PR DESCRIPTION
**Pull request by snazy**
Take 2, using GitHub's native functionality.
Basically reverts #3004.

See https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#concurrency

### CODE REVIEW

**Review by kbendick — COMMENTED** [review]
Thanks for tagging me @snazy. I looked into this when I made the original cancel duplicate workloads file, but it was pretty beta at the time and I couldn't get the `group` such that it would cancel the same PR (I didn't try that hard as it was pretty new functionality and the apache/spark repo uses the other project).

Have you tested this configuration in your own repo, or in any other repo? Does it work the same?

**Comment by kbendick** [comment]
Ahhh I see the docs have been updated: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-only-cancel-in-progress-jobs-or-runs-for-the-current-workflow

My (very limited) experience was that it would cancel any workflow (outside of the current PR or not), but I guess if you have `github.ref` then that should be ok.

This has a cautious +1 from me, but if you've used this elsewhere then I definitely trust you. This looks to cover all of the workflows that were part of the old cancel-duplicates.

**Review by kbendick — APPROVED** [review]
Gonna give this a cautious +1 for the reasons mentioned.

If you've used this concurrency group and it worked as expected (cancels duplicates from the same branch), then I'm good with it. Like I said, when I tried it before it was _very_ beta but it would be nice to use the native functionality.

If you don't have experience with it, we should maybe test it quickly in a fork or something (or have a committer on hand to revert).

**Comment by kbendick** [comment]
Looking at this Stack Overflow post, I'm much more confident in this now: https://stackoverflow.com/questions/58033366/how-to-get-the-current-branch-within-github-actions

**Comment by snazy** [comment]
We've merged the same change into Nessie today and I realized that Quarkus is also using this technique now - they have more complex conditions though: https://github.com/quarkusio/quarkus/blob/main/.github/workflows/ci-actions-incremental.yml#L36-L38

**Comment by snazy** [comment]
Realized that the proposed change would cancel previous CI runs on the master branch as well, which is rather not so good.

**Review by nastra — APPROVED** [review]

File: .github/workflows/flink-ci.yml [hunk 1]
```diff
@@ -59,6 +59,10 @@ on:
     - 'README.md'
     - 'CONTRIBUTING.md'
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
```
Comment (kbendick) [inline .github/workflows/flink-ci.yml:64]: Good call 👍

Comment (snazy) [reply .github/workflows/flink-ci.yml:64]: Yea, no event without a name (I hope so ;) )

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks for fixing this, @snazy!

**Pull request merged**